File: UTB-AP5PW-Invoicer.Client/src/types/payment.ts
export interface Payment {
  id: number;
  invoiceId: number;
  amount: number;
  paymentDate: string;
  paymentMethod: string;
}

export interface CreatePaymentDto {
  invoiceId: number;
  amount: number;
  paymentDate: string;
  paymentMethod: string;
}

export interface UpdatePaymentDto extends CreatePaymentDto {
  id: number;
}

File: UTB-AP5PW-Invoicer.Domain/Entities/Payment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UTB_AP5PW_Invoicer.Domain.Entities
{
    public class Payment : Entity<int>
    {
        [ForeignKey(nameof(Invoice))]
        public int InvoiceId { get; set; }

        [Required]
        public DateTimeOffset PaymentDate { get; set; }

        [Required]
        public decimal Amount { get; set; }

        [Required]
        public string PaymentMethod { get; set; }

        public Invoice Invoice { get; set; }
    }
}
